In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv("/Users/janet.xuishi/Documents/DA_L7/CA2.0_stats_garments_worker_productivity.csv")

In [3]:
#df.head()

In [4]:
#df.describe()

# Stats CA2.1

Inferential Statistics - part 1

The company believes implemented new machines and they believe this would reduce the average
overtime (in minutes) that the workers will do, which will lead to a cost reduction in the end. They tried it in
a group of 200 workers, which shows an average overtime of 6300 minutes. Can we say that the company
is right? Formulate an appropriate hypothesis and justify your answer.

Number of workers total population in csv file: 36240
Experiment ran among sample of 200 workers, result of after new machines implementation overtime MEAN is 6300 minutes.
We need to find out if the MEAN overtime is higher or lower than 6300 minutes to determine if implementing the new machines is a success or failure.

H0 New machines did not reduce the over time, before and after MEAN over time is the same, or after over time is greater than the old machines overtime.
H1 New machines reduced MEAN over time before, less than old machines MEAN over time.

In [5]:
# Calculte the MEAN and STD of the overtime for the entire population
population_mean_overtime = round(df["over_time"].mean(),2)
population_std = round(df["over_time"].std(),2)
print(population_mean_overtime, population_std)

6508.21 2864.51


In [6]:
# Given the mean and std of the overtime of the sample (200 workers and the average overtime among 200 workers after new machines implementation )
sample_mean = 6300
sample_size = 200

In [7]:
# Calculte standard error 
standard_error = round(population_std / (sample_size ** 0.5),2)
print(standard_error)

202.55


Hypothesis: 

H0 New machines did NOT reduce the over time, before and after MEAN over time is the same, or greater than the old machines overtime.

H1 New machines reduced MEAN over time before, less than old machines MEAN over time.

In [8]:
#import scipy.stats libary
from scipy.stats import norm
#calulate z-score
z_score = ( sample_mean - population_mean_overtime) / ( population_std / ( sample_size ** 0.5 ))
print(f"z_score is: {z_score}")

#calulate p-value
p_value = 2 * norm.sf(abs(z_score))
print(f"p_value is: {p_value}")


z_score is: -1.027936386403616
p_value is: 0.30397975216140327


Conclusion: 

Z-score is negative: -1.03, indicates that sample mean 6300 minutes is less than population mean.

P-value is 0.30 which is much higher than alpha level of 0.05, indicates the difference in mean overtime, we do NOT have sufficient evidence to reject H0. The data does NOT provide strong evidence to support company's claim that implementation of new machines was successful in reducing overtime. In the real manufacturing world, there are many factors to consider, we don't have enough evidence in this data file to reject H0 doesn't mean it's true. Implementation of new machines usually involves testing, production line halt due to errors or adjustment, worker operation fault etc, these data are not captured in this data file.